### PR TITLE
searchByPhoneCode Added

### PR DESCRIPTION
**Pull request by sauravmankotia**
- search by phoneCode
- search including "+"

### CODE REVIEW

**Comment by knowable-noujima-akinori** [comment]
@Daniel-Ioannou i think this function should be include in the lib, please help to review on this PR

**Comment by uyit14** [comment]
@Daniel-Ioannou  Please look into this PR

File: lib/src/country.dart [hunk 1]
```diff
@@ -119,14 +119,30 @@ class Country {
...
+        phoneCode.startsWith(query.toLowerCase()) ||
+            name.toLowerCase().startsWith(query.toLowerCase()) ||
+            countryCode.toLowerCase().startsWith(query.toLowerCase()) ||
+            (localizations
+                ?.countryName(countryCode: countryCode)
+                ?.toLowerCase()
+                .startsWith(query.toLowerCase()) ??
+                false);
+    }
+  }
```
Comment (Daniel-Ioannou) [inline lib/src/country.dart:145]: Please simplified. You wrote the "Search condition" twice. 
```suggestion
  bool startsWith(String query, CountryLocalizations? localizations) {
    String _query = query;
    if (query.startsWith("+")) {
      _query = query.replaceAll("+", "").trim();
    }

      return phoneCode.startsWith(_query.toLowerCase()) ||
          name.toLowerCase().startsWith(_query.toLowerCase()) ||
          countryCode.toLowerCase().startsWith(_query.toLowerCase()) ||
          (localizations
              ?.countryName(countryCode: countryCode)
              ?.toLowerCase()
              .startsWith(_query.toLowerCase()) ??
              false);
    }
```

**Review by Daniel-Ioannou — CHANGES_REQUESTED** [review]

**Comment by Daniel-Ioannou** [comment]
Feature added at #78 and released in [v2.0.14](https://pub.dev/packages/country_picker/versions/2.0.14).

**Pull request closed without merging**